scripts/generate_llms_by_category.py

```diff
@@ -0,0 +1,154 @@
+os.makedirs(output_dir, exist_ok=True) # make the directory if it doesn't exist
+
+# Extracts and writes a per-category LLMS file 
+def extract_category(category, shared_data=None): 
+    with open(llms_input_path, 'r', encoding='utf-8') as f: # read the full LLMS input file
+        llms = f.read() 
+
+    blocks = re.findall(
+        r"Doc-Content: (.*?)\n--- BEGIN CONTENT ---\n(.*?)\n--- END CONTENT ---", # extract all documentation blocks
+        llms, re.DOTALL
+    )
+
+    index_lines = [] 
+    content_blocks = [] 
+
+    for url, content in blocks: 
+        metadata_match = re.search(r"---\n(.*?)\n---", content, re.DOTALL) # extract the metadata block to find categories
+        if not metadata_match:
+            continue
+
+        metadata = metadata_match.group(1)
+
+        category_line = re.search(r"categories:\s*(.*)", metadata) # looks for a line starting with 'categories:' in the metadata
+        if not category_line:
+            continue
+
+        def infer_section_label(url): # classify the doc page based on URL path
+            for section in SECTION_PRIORITY:
+                if f"/{section}/" in url:
+                    return section
+            return "other"
+
+        # Check if the given category is listed in the metadata
+        tags = [tag.strip().lower() for tag in category_line.group(1).split(',')] # splits tags by comma 
+        if category.lower() in tags:
+            section_label = infer_section_label(url)
+            #raw_url = f"{RAW_BASE_URL}{url}"  # `url` already includes a leading slash
+            # Fix: Convert /docs/.../page -> relative GitHub path
+            if "/docs/" in url:
+                rel_path = url.split("/docs/")[1].rstrip("/") + ".md"
+                raw_url = f"{RAW_BASE_URL}/{rel_path}"
+            else:
+                raw_url = url  # fallback
+            index_lines.append(f"Doc-Page: {raw_url} [type: {section_label}]")
+            content_blocks.append(f"Doc-Content: {url}\n--- BEGIN CONTENT ---\n{content.strip()}\n--- END CONTENT ---") # store full page
+
+    if not content_blocks: # if no pages matched the category, skip 
+        print(f"[!] Skipping {category} – no matching pages.")
+        return
+
+    # Output file for this category
+    output_file = os.path.join(output_dir, f"{category.lower()}-llms.txt") # write to output file
+    with open(output_file, 'w', encoding='utf-8') as f:
+
+        # Intro context block to help LLMs understand purpose of the file
+        f.write(f"# {PROJECT_NAME} Developer Documentation (LLMS Format)\n\n")
+        f.write(f"This file contains documentation for {PROJECT_DESCRIPTION}\n")
+        f.write("It is intended for use with large language models (LLMs) to support developers working with Wormhole. The content includes selected pages from the official docs, organized by product category and section.\n\n")
+        f.write(f"This file includes documentation related to the product: {category}\n\n")
```

Comment:
```suggestion
        f.write(f"This file includes documentation related to the category: {category}\n\n")
```

Happened to notice the following was returned, and "Reference" isn't a product, so figured we should generalize this so AI doesn't think Reference and Basics are products

> This file includes documentation related to the product: Reference

Reply:
when we concatenate these two to the category files this is not written for their part, they have a different intro 
it is only added to their individuals because that's the template to build the page and I needed to do that before being able to break it down and add it - I wrote product to emphasize all the other llms products

tbh I thought we wouldn't show reference as singular downloadable llms, its just a few pages and they are added to every file anyway - it's there just bc I need to generate it in order to add it 
I would prefer either removing reference from downloadable or find another way around in the script to keep saying all the others are products instead of calling the products categories 

cc @martin0995 wdyt

Reply:
I think it's important to define the products so it doesn't get confused with the basics etc in the same file, there's a line somewhere that tells the ai to focus its answers on the products unless asked otherwise

Reply:
@eshaben ok nvm the book I wrote in the comments aha I fixed in the script to write two different things even in the file :)) check again